esp/services/ws_fs/ws_fsService.cpp

```diff
@@ -2800,54 +2808,62 @@ bool CFileSprayEx::onFileList(IEspContext &context, IEspFileListRequest &req, IE
     return true;
 }
 
-
-void CFileSprayEx::queryDropZoneInfo(double clientVersion, const char* dropZone, bool ECLWatchVisibleOnly, IArrayOf<IConstTpDropZone>& dropZones)
+bool CFileSprayEx::checkDropZoneIPAndPath(double clientVersion, const char* dropZoneName, const char* netAddr, const char* path)
 {
-    CTpWrapper tpWrapper;
-    tpWrapper.getTpDropZones(clientVersion, dropZone, ECLWatchVisibleOnly, dropZones);
-    if (dropZones.empty())
-        throw MakeStringException(ECLWATCH_INVALID_INPUT, "DropZone %s not found.", dropZone);
+    if (isEmptyString(netAddr) || isEmptyString(path))
+        throw MakeStringException(ECLWATCH_INVALID_INPUT, "NetworkAddress or Path not defined.");
+
+    IpAddress ipToMatch;
+    ipToMatch.ipset(netAddr);
+    if (ipToMatch.isNull())
+        throw MakeStringException(ECLWATCH_INVALID_INPUT, "Invalid network address %s.", netAddr);
+
+    Owned<IEnvironmentFactory> envFactory = getEnvironmentFactory();
+    Owned<IConstEnvironment> constEnv = envFactory->openEnvironment();
+    if (!isEmptyString(dropZoneName))
+    {
+        Owned<IConstDropZoneInfo> dropZoneInfo = constEnv->getDropZone(dropZoneName);
+        if (!dropZoneInfo)
+            throw MakeStringException(ECLWATCH_INVALID_INPUT, "DropZone %s not found.", dropZoneName);
+
+        return checkDropZoneIPAddressAndPath(clientVersion, dropZoneInfo, ipToMatch, path);
+    }
+
+    Owned<IConstDropZoneInfoIterator> dropZoneItr = constEnv->getDropZoneIterator();
+    ForEach(*dropZoneItr)
+    {
+        IConstDropZoneInfo& dropZoneInfo = dropZoneItr->query();
+        if (checkDropZoneIPAddressAndPath(clientVersion, &dropZoneInfo, ipToMatch, path))
+            return true;
+    }
```

Comment:
See previous comment, this code would be a lot simpler/more readable if calling a getDropZonesByIP() method.